onnx/shape_inference/implementation.cc

```diff
@@ -77,6 +87,39 @@ void checkShapesAndTypes(const TypeProto_Tensor& inferredType, const TypeProto_T
   }
 }
 
+void checkShapesAndTypes(const TypeProto_SparseTensor& inferredType, const TypeProto_SparseTensor& existingType) {
```

Comment:
Can this be simplified by using a template?

Reply:
Will do.